src/Router.sol

```diff
@@ -198,6 +202,28 @@ contract Router is IRouter, EIP712, FeeGenerator {
         _execute(user, logics, tokensReturn, referralCode);
     }
 
+    /// @notice Execute permitted execution data through the given user's agent. Creates an agent for user if not created.
+    ///         Fees are charged in the user's agent based on the scenarios defined in the FeeGenerator contract, which
+    ///         calculates fees by logics and msg.value.
+    /// @param details The execution details permitted by user
+    /// @param user The user address
+    /// @param signature The user's signature bytes
+    function executeBySig(
+        IParam.ExecutionDetails calldata details,
+        address user,
+        bytes calldata signature
+    ) external payable whenReady(user) {
+        uint256 deadline = details.deadline;
+        uint256 nonce = details.nonce;
+        // Verify deadline, signature and nonce
+        if (block.timestamp > deadline) revert SignatureExpired(deadline);
+        if (!user.isValidSignatureNow(_hashTypedDataV4(details._hash()), signature)) revert InvalidSignature();
+        if (executionNonces[user] != nonce) revert InvalidNonce();
+        executionNonces[user] = nonce;
```

Comment:
```suggestion
        executionNonces[user] = ++nonce;
```
Should we increment the `nonce` by 1?

Reply:
Ah yes. Let me update the test to check that also.